ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template7.kt

```diff
@@ -0,0 +1,668 @@
+import com.revenuecat.purchases.ui.revenuecatui.UIConstant
+import com.revenuecat.purchases.ui.revenuecatui.composables.Footer
+import com.revenuecat.purchases.ui.revenuecatui.composables.IntroEligibilityStateView
+import com.revenuecat.purchases.ui.revenuecatui.composables.Markdown
+import com.revenuecat.purchases.ui.revenuecatui.composables.OfferDetails
+import com.revenuecat.purchases.ui.revenuecatui.composables.PaywallIcon
+import com.revenuecat.purchases.ui.revenuecatui.composables.PaywallIconName
+import com.revenuecat.purchases.ui.revenuecatui.composables.PurchaseButton
+import com.revenuecat.purchases.ui.revenuecatui.composables.RemoteImage
+import com.revenuecat.purchases.ui.revenuecatui.composables.SelectedTierView
+import com.revenuecat.purchases.ui.revenuecatui.composables.StatusBarSpacer
+import com.revenuecat.purchases.ui.revenuecatui.composables.TierSwitcher
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallState
+import com.revenuecat.purchases.ui.revenuecatui.data.PaywallViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.isInFullScreenMode
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.ProcessedLocalizedConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.TemplateConfiguration
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.localizedDiscount
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.MockViewModel
+import com.revenuecat.purchases.ui.revenuecatui.data.testdata.TestData
+import com.revenuecat.purchases.ui.revenuecatui.extensions.conditional
+import com.revenuecat.purchases.ui.revenuecatui.extensions.introEligibility
+import com.revenuecat.purchases.ui.revenuecatui.extensions.packageButtonActionInProgressOpacityAnimation
+import com.revenuecat.purchases.ui.revenuecatui.extensions.packageButtonColorAnimation
+import com.revenuecat.purchases.ui.revenuecatui.helpers.ResourceProvider
+import com.revenuecat.purchases.ui.revenuecatui.helpers.shouldUseLandscapeLayout
+
+private object Template7UIConstants {
+    val featureIconSize = 25.dp
+    val checkmarkSize = 18.dp
+    val discountPadding = 8.dp
+    const val headerAspectRatio = 2f
+}
+
+@Composable
+internal fun Template7(
+    state: PaywallState.Loaded,
+    viewModel: PaywallViewModel,
+) {
+    val packagesConfig = state.templateConfiguration.packages
+    require(packagesConfig is TemplateConfiguration.PackageConfiguration.MultiTier) {
+        // This should never happen
+        "The configuration is not MultiTier"
+    }
+
+    val firstTier = packagesConfig.firstTier
+    val allTiers = packagesConfig.allTiers
+
+    var packageSelectorVisible by remember {
+        mutableStateOf(state.templateConfiguration.mode != PaywallMode.FOOTER_CONDENSED)
+    }
+
+    var selectedTier by remember {
+        mutableStateOf(firstTier)
+    }
+
+    val colorForTier = state.templateConfiguration.getCurrentColorsForTier(tier = selectedTier)
+
+    Column(
```

Comment:
Wonder if we should also try to animate all the main content when switching tiers for a smoother transition since everything can pretty much change... Might be slightly tricky though, depending on what we're animating...

Reply:
Added some animations and it feels soooooo much better!